utils/integration-test-utils/src/up_client_failing_register.rs

```diff
@@ -0,0 +1,85 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use async_trait::async_trait;
+use log::debug;
+use std::sync::atomic::AtomicU64;
+use std::sync::Arc;
+use up_rust::{UCode, UListener, UMessage, UStatus, UTransport, UUri};
+
+pub struct UPClientFailingRegister {
```

Comment:
Can you check into using a mocking library?

I wonder if there's a better way of going about this than making an entirely new UTransport implementation.

Can we use this [MockTransport](https://github.com/eclipse-uprotocol/up-rust/blob/9589ea8ced5ece716a06e52fdca4f8240e1dc8f6/src/utransport.rs#L186) from up-rust?

Reply:
Will open an issue for this

Reply:
https://github.com/eclipse-uprotocol/up-rust/issues/185